WordPress/Classes/Services/AccountSettingsService.swift

```diff
@@ -1,34 +1,127 @@
 import Foundation
+import Reachability
 import RxCocoa
 import RxSwift
 
 let AccountSettingsServiceChangeSaveFailedNotification = "AccountSettingsServiceChangeSaveFailed"
 
-struct AccountSettingsService {
-    let remote: AccountSettingsRemote
+protocol AccountSettingsRemoteInterface {
+    var settings: Observable<AccountSettings> { get }
+    func updateSetting(change: AccountSettingsChange, success: () -> Void, failure: ErrorType -> Void)
+}
+
+extension AccountSettingsRemote: AccountSettingsRemoteInterface {}
+
+class AccountSettingsService {
+    struct Defaults {
+        static let stallTimeout = 4.0
+        static let maxRetries = 3
+        static let pollingInterval = 60.0
+    }
+
+    let remote: AccountSettingsRemoteInterface
     let userID: Int
 
     private let context = ContextManager.sharedInstance().mainContext
 
-    init(userID: Int, api: WordPressComApi) {
-        self.remote = AccountSettingsRemote(api: api)
-        self.userID = userID
+    var testScheduler: SchedulerType? = nil
+    private var scheduler: SchedulerType {
+        return testScheduler ?? MainScheduler.instance
     }
 
-    func refreshSettings(completion: (Bool) -> Void) {
-        remote.getSettings(
-            success: {
-                (settings) -> Void in
+    convenience init(userID: Int, api: WordPressComApi) {
+        let remote = AccountSettingsRemote.remoteWithApi(api)
+        self.init(userID: userID, remote: remote)
+    }
 
-                self.updateSettings(settings)
-                completion(true)
-            }, failure: {
-                (error) -> Void in
+    init(userID: Int, remote: AccountSettingsRemoteInterface) {
+        self.userID = userID
+        self.remote = remote
+    }
 
-                DDLogSwift.logError(String(error))
-                completion(false)
+    var testReachability: Observable<Bool>? = nil
```

Comment:
Ah yes, now I remember why we got away with doing host-based reachability instances.